src/driver/xdma_sgdma.h

```diff
@@ -67,16 +67,36 @@ int hpdev_init_channels(struct hermes_pci_dev *hpdev);
 ssize_t xdma_channel_read_write(struct xdma_channel *chnl,
 		struct iov_iter *iter, loff_t pos);
 
+static inline struct xdma_channel *xdma_get_chnl(struct xdma_channel *channels,
+		struct ida_wq *ida_wq)
```

Comment:
This is all rather messy. Why is everything static inline functions in seemingly random different headers. Much of this doesn't seem like it needs to be static inline in a header.

Make a couple clear API functions in the headers: xdma_[get|release]_[h2c|c2h]() and everything else should be static functions in a single C file.

I understand you're dealing with a mess caused by the xdma driver, but we should be able to do better and slowly clean it up. Maybe someday this will all be cleaned up to the extent that it can all be in one C file.